From ff8ecbe8891d592e645927659318720f9e190054 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sun, 6 Feb 2022 17:23:46 +0100
Subject: [PATCH] [PATCH] Consider pkgconfig sysroot for pkgdatadir

For libs/cflags this is done automatically, but not for manually accessed
variables. This matches what wayland-protocols does.
---
 src/meson.build           | 6 +++---
 src/wayland-client.pc.in  | 2 +-
 src/wayland-scanner.pc.in | 2 +-
 src/wayland-server.pc.in  | 2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index 61edbc8..d621efd 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -50,7 +50,7 @@ if get_option('scanner')
 		version: meson.project_version(),
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name()),
+			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name()),
 			'bindir=' + join_paths('${prefix}', get_option('bindir')),
 			'wayland_scanner=wayland-scanner'
 		],
@@ -179,7 +179,7 @@ if get_option('libraries')
 		filebase: 'wayland-server',
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name())
+			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name())
 		]
 	)
 
@@ -211,7 +211,7 @@ if get_option('libraries')
 		filebase: 'wayland-client',
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name())
+			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name())
 		]
 	)

-- 
2.32.0

